test/jest/__tests__/utils/utils.DependencyUtils.ts.spec.ts

```diff
@@ -0,0 +1,280 @@
+    categories?: string[]
+}): ThunderstoreCombo {
+    const combo = new ThunderstoreCombo();
+
+    const mod = new ThunderstoreMod();
+    mod.setName(`${params.name}`);
+    mod.setFullName(`author-${params.name}`);
+    mod.setDependencies(params.dependencies || []);
+    mod.setCategories(params.categories || []);
+    combo.setMod(mod);
+
+    const version = new ThunderstoreVersion();
+    version.setVersionNumber(new VersionNumber(params.versionNumber));
+    version.setName(`author-${params.name}-${params.versionNumber}`);
+    version.setDependencies(params.dependencies || []);
+    combo.setVersion(version);
+
+    return combo;
+}
+
+let mod: ThunderstoreCombo;
+let modWithDependency: ThunderstoreCombo;
+let modWithDependency2: ThunderstoreCombo;
+let modWithTheSameDependency: ThunderstoreCombo;
+let modWithTwoDependeciesThatHaveTheSameDependency: ThunderstoreCombo;
+let modWithDeepDependency: ThunderstoreCombo;
+let modPack: ThunderstoreCombo;
+let modWithLoopDependencyA: ThunderstoreCombo;
+let modWithLoopDependencyB: ThunderstoreCombo;
+
+let game: Game;
+
+const beforeSetup = () => {
+
+    game = new Game("", "RiskOfRain2", "", "", [], "", "", [], "",
+        GameSelectionDisplayMode.VISIBLE, GameInstanceType.GAME, PackageLoader.BEPINEX, []);
+
+    mod = createCombo({
+        name: "mod",
+        versionNumber: "1.0.0"
+    });
+
+    modWithDependency = createCombo({
+        name: "modWithDependency",
+        versionNumber: "1.0.0",
+        dependencies: ["author-mod-1.0.0"]
+    });
+
+    modWithDependency2 = createCombo({
+        name: "modWithDependency2",
+        versionNumber: "1.0.0",
+        dependencies: ["author-mod-1.0.0"]
+    });
+
+    modWithTheSameDependency = createCombo({
+        name: "modWithTheSameDependency",
+        versionNumber: "1.0.0",
+        dependencies: ["author-mod-1.0.0"]
+    });
+
+    modWithTwoDependeciesThatHaveTheSameDependency = createCombo({
+        name: "modWithTwoDependeciesThatHaveTheSameDependency",
+        versionNumber: "1.0.0",
+        dependencies: ["author-modWithDependency-1.0.0", "author-modWithTheSameDependency-1.0.0"]
+    });
+
+    modWithDeepDependency = createCombo({
+        name: "modWithDeepDependency",
+        versionNumber: "1.0.0",
+        dependencies: ["author-modWithDependency-1.0.0"]
+    });
+
+    modPack = createCombo({
+        name: "modPack",
+        versionNumber: "1.0.0",
+        dependencies: ["author-mod-1.0.0"],
+        categories: ["Modpacks"]
+    });
+
+    modWithLoopDependencyA = createCombo({
+        name: "modWithLoopDependencyA",
+        versionNumber: "1.0.0",
+        dependencies: ["author-modWithLoopDependencyB-1.0.0"]
+    });
+
+    modWithLoopDependencyB = createCombo({
+        name: "modWithLoopDependencyB",
+        versionNumber: "1.0.0",
+        dependencies: ["author-modWithLoopDependencyA-1.0.0"]
+    });
+};
+
+describe("DependencyUtils.getFullDependencyList", () => {
+
+    beforeEach(() => {
```

Comment:
Should this be `beforeAll`? I don't think `beforeSetup` currently does anything that needs to be redone/reset before each test case.